server/src/tasks/deposit.js

```diff
@@ -0,0 +1,43 @@
+const mongoose = require('mongoose')
+const Transfer = mongoose.model('Transfer')
+const { lockAccount, withAccount } = require('@utils/account')
+const { createNetwork } = require('@utils/web3')
+const utils = require('@utils/token')
+const bridgeUtils = require('@utils/bridge')
+const { handleReceipt } = require('@handlers/receipts')
+
+const relayTokens = withAccount(async (account, { transferId }) => {
+  const transferDoc = await Transfer.findById(transferId)
+  if (!transferDoc || transferDoc.status !== 'READY') {
+    return
+  }
+  try {
+    const network = createNetwork(transferDoc.bridgeType, account)
+    const approveReceipt = await utils.approve(network, transferDoc.toObject())
```

Comment:
need to check allowance before the approve. Most of the transactions are sent from the backend wallets, so in practice we'll call approve only several times